src/main/kotlin/study/racing/inputview/CarNameBasedInputView.kt

```diff
@@ -0,0 +1,30 @@
+package study.racing.inputview
+
+import java.lang.IllegalArgumentException
+
+class CarNameBasedInputView : InputView<CarNameBasedInputView.CarNameBasedRacingInputResult> {
+
+    override fun receive(): CarNameBasedRacingInputResult {
+        println("경주할 자동차 이름을 입력하세요(이름은 쉼표(,)를 기준으로 구분)")
+        val carNames = readLine()!!.toString()
```

Comment:
`!!`은 null이 아니라는 확신이 있을 때만 쓰시는 걸 추천드리고, 왠만해선 실제 코드에는 사용하지 않는 것을 권장드려요.
왜냐하면 코틀린은 nullable 타입을 사용하면 컴파일러가 null 체크를 요구하고, null에 대한 안전한 처리를 유도합니다. 하지만 `!!` 연산자를 사용하면 nullable 타입 안전성을 해칠 수 있어요.(코드 가독성을 저하시키기도 하구요)

```suggestion
        val carNames = readln()
        // 또는 
        val carNames = readLine() ?: throw IlegalArgumentException("null은 입력될 수 없습니다")
```